twill-yarn/src/main/hadoop21/org/apache/twill/internal/yarn/Hadoop21YarnAppClient.java

```diff
@@ -127,6 +135,7 @@ private void addRMToken(ContainerLaunchContext context) {
       return;
     }
 
+    YarnClient yarnClient = createYarnClient();
```

Comment:
This `addRMToken` method only gets called from the `ApplicationSubmitter.submit()` method, which only get triggered form the `createLauncher` method. Since there is already a new `YarnClient` created inside the `createLauncher` method, you can just have that one passed to this `addRMToken` method and use it instead of creating another one here.